src/gui/DatabaseWidget.cpp

```diff
@@ -1520,13 +1524,30 @@ void DatabaseWidget::onGroupChanged()
 
 void DatabaseWidget::onDatabaseModified()
 {
-    if (!m_blockAutoSave && config()->get(Config::AutoSaveAfterEveryChange).toBool()) {
+    refreshSearch();
+    int autosaveDelayMs = m_db->metadata()->autosaveDelayMin() * 60 * 1000; // min to msec for QTimer
+    bool autosaveAfterEveryChangeConfig = config()->get(Config::AutoSaveAfterEveryChange).toBool();
+    if (m_autosaveTimer->isActive() || (autosaveDelayMs > 0 && autosaveAfterEveryChangeConfig)) {
+        // reset delay when modified
+        m_autosaveTimer->start(autosaveDelayMs);
```

Comment:
Here the only place where you start the timer. Where is the stop call? For example, if user saved the DB manually, should we force the stop of timer? I mean the autosaving should not happen by slot triggering on line 1543 if the DB is already saved.

Reply:
Are you refering to the next timer cycle? I set the timer on [line 226](https://github.com/keepassxreboot/keepassxc/pull/9100/files#diff-742a7522409932dea5fb69302a8baf103ea3cb52b01653ae6f570e6b58ced3c4R226) to be a [single shot timer](https://doc.qt.io/qt-6/qtimer.html#singleShot-prop)
`m_autosaveTimer->setSingleShot(true);`
Therefor it will only "fire" once. There should not be any need to stop it.

Reply:
I mean the next workflow:
- assume the delay is configured for 15 min.
- User made all needed changes in 10 min and decided to save it manually
- In another 5 min the timer got a timeout and triggered the slot function which will try to save the DB again.
Does it make sense?

Reply:
Yea, you are right this makes sense, i should handle this case, will add this

Reply:
Fixed at [33125e1](https://github.com/keepassxreboot/keepassxc/pull/9100/commits/33125e1a603598f36148c48775f77d85268d6277)